infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -137,25 +140,29 @@ private void runOrchestrator() {
                 log.warn("Unable to acquire manager status: ", e);
             }
             try {
-                if (managerStatus != null) {
-                    if (managerStatus.getStatus() == StatusType.FAILED || managerStatus.getStatus() == StatusType.COMPLETED) {
-                        checkpointerJvmManager.shutdown();
-                    } else if (managerStatus.getStatus() == StatusType.STARTED && !checkpointerJvmManager.isRunning()
-                            && !checkpointerJvmManager.isInvoked()) {
-                        checkpointerJvmManager.invokeCheckpointing();
-                    }
-                }
-
                 if (isLeader) {
                     if (managerStatus != null && managerStatus.getStatus() == StatusType.STARTED) {
-                        getCompactorLeaderServices().validateLiveness();
+                        if (distributedCheckpointerHelper.isCompactionDisabled()) {
```

Comment:
on all normal runs this will result in 2 RPC - one to check and the other to validate. This can be combined into one RPC if we just wrap these two in a single transaction right?
But this is a minor optimization and please consider it optional